GooglePlayInstant/Editor/PlayInstantQuickDeployHttpHandler.cs

```diff
@@ -0,0 +1,361 @@
+        /// <param name="postHeaders"> A dictionary representing a set of key-value pairs to be added to the request headers</param>
+        /// <returns></returns>
+        public static string SendHttpPostRequest(string endpoint, Dictionary<string, string> postForm,
+            Dictionary<string, string> postHeaders)
+        {
+            var form = new WWWForm();
+            if (postForm != null)
+            {
+                foreach (var pair in postForm)
+                {
+                    form.AddField(pair.Key, pair.Value);
+                }
+            }
+
+            return SendHttpPostRequest(endpoint, postForm != null ? form.data : null, postHeaders);
+        }
+
+        /// <summary>
+        /// Sends a general GET request to the specified endpoint along with specified parameters and headers
+        /// </summary>
+        /// <param name="endpoint">The endpoint to with the get request should be sent</param>
+        /// <param name="getParams"> A dictionary representing ke-value pairs that should be attached to the endpoint as GET parameters</param>
+        /// <param name="getHeaders">A dictionary representing key-value pairs that constitude the data to be added to the request headers</param>
+        /// <returns></returns>
+        public static string SendHttpGetRequest(string endpoint, Dictionary<string, string> getParams,
+            Dictionary<string, string> getHeaders)
+        {
+            var fullEndpoint = endpoint + "?";
+            var form = new WWWForm();
+            if (getHeaders != null)
+            {
+                foreach (var pair in getHeaders)
+                {
+                    form.headers.Add(pair.Key, pair.Value);
+                }
+            }
+
+            if (getParams != null)
+            {
+                foreach (var pair in getParams)
+                {
+                    fullEndpoint += pair.Key + "=" + pair.Value + "&";
+                }
+            }
+
+            var www = new WWW(fullEndpoint, null, form.headers);
+            return GetResultWhenDone(www);
+        }
+
+
+        // Waits until the WWW request is done and returns the results.
+        private static string GetResultWhenDone(WWW www)
+        {
+            while (!www.isDone)
+            {
+                Debug.LogFormat("Progress: {0}/100 completed", (www.progress * 100).ToString());
+            }
+
+            return www.text;
+        }
+    }
+
+
+    /// <summary>
+    /// A class representing a server with the sole purpose of providing an endpoint that listens for the autorization
+    /// code from google's OAuth2 API's authorization page, on which the user grants an application access their data
+    /// in a given scope.
+    ///
+    /// <see cref="https://developers.google.com/identity/protocols/OAuth2"/> for an overview of OAuth2 protocol for
+    /// Google APIs.
+    /// </summary>
+    public class QuickDeployOAuth2CallbackEndpointServer
```

Comment:
I'm having a hard time so far understanding why this is needed and what the alternatives might be, in part because I don't see it used anywhere yet. Maybe we should chat on GVC about the approach.